src/components/Fields/DateField.js

```diff
@@ -0,0 +1,165 @@
+import React, { Component } from 'react';
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import StyleObjectPropType from '../../prop-types/style';
+import TextField from './TextField';
+import DateInlinePicker from './DateInlinePicker';
+import Tooltip from '../Tooltip/Tooltip';
+import { TooltipPosition } from '../Tooltip/TooltipEnums';
+import { DateFieldRangePosition } from './TextFieldEnums';
+import './DateField.scss';
+
+const propTypes = {
+    className: PropTypes.string,
+    forceHideCalendar: PropTypes.bool,
+    isRange: PropTypes.bool,
+    maxDate: PropTypes.instanceOf(Date),
+    minDate: PropTypes.instanceOf(Date),
+    onBlur: PropTypes.func,
+    onChange: PropTypes.func,
+    onFocus: PropTypes.func,
+    rangePosition: PropTypes.oneOf([
+        DateFieldRangePosition.START,
+        DateFieldRangePosition.FINISH,
+    ]),
+    style: StyleObjectPropType(),
+    value: PropTypes.oneOfType([
+        PropTypes.instanceOf(Date),
+        PropTypes.arrayOf(PropTypes.instanceOf(Date)),
+    ]),
+};
+
+const defaultProps = {
+    className: '',
+    isRange: false,
+    forceHideCalendar: false,
+    maxDate: null,
+    minDate: null,
+    onBlur: null,
+    onChange: null,
+    onFocus: null,
+    rangePosition: DateFieldRangePosition.START,
+    style: null,
+    value: null,
+};
+
+class DateField extends Component {
+    state = {
+        selectedDate: this.props.value,
+        showDatePicker: false,
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (this.props.value !== nextProps.value) {
+            this.setState({ selectedDate: nextProps.value });
+        }
+        if (this.props.forceHideCalendar !== nextProps.forceHideCalendar) {
+            if (nextProps.forceHideCalendar) {
+                this.setState({ showDatePicker: false });
+            }
+        }
+    }
+
+    clearInputAfterChange = (inputValue) => {
+        if (inputValue === null || inputValue === '') {
+            this.setState({ selectedDate: null });
+        }
+    }
+
+    datePickerClose() {
+        if (this.props.isRange === false ||
+            (this.state.selectedDate && this.state.selectedDate.length === 2)
+        ) {
+            this.setState({
+                showDatePicker: false,
+            });
+        }
+    }
+
+    datePickerOpen() {
+        this.setState({
+            showDatePicker: true,
+        });
+    }
+
+    formatDate = (date) => {
+        if (!date) {
+            return date;
```

Comment:
Might be better to return something explicit here rather - e.g. `return undefined`

Reply:
Good call... I've refactored – will discuss my suggested changes offline 👍